util/answer.go

```diff
@@ -13,3 +21,30 @@ func GenerateAnonymousAvatar(alias string) string {
 	}
 	return fmt.Sprintf("https://api.dicebear.com/9.x/thumbs/svg?seed=%s", match)
 }
+
+func generateUniqueAlias(db *gorm.DB) (string, error) {
+	// Generate cryptographically secure random index
+	nameIndex, err := rand.Int(rand.Reader, big.NewInt(int64(len(names))))
+	if err != nil {
+		return "", fmt.Errorf("failed to generate random name index: %w", err)
+	}
+
+	name := names[nameIndex.Int64()]
+	var lastUser models.User
+	pattern := fmt.Sprintf("%s_%%", name)
+	result := db.Where("alias LIKE ?", pattern).Order("created_at DESC").First(&lastUser)
+
+	nextNum := 1
+	if result.Error == nil {
+		re := regexp.MustCompile(fmt.Sprintf(`^%s_(\d+)$`, name))
+		if matches := re.FindStringSubmatch(lastUser.Alias); len(matches) == 2 {
+			if n, err := strconv.Atoi(matches[1]); err == nil {
+				nextNum = n + 1
+			}
+		}
+	} else if result.Error != gorm.ErrRecordNotFound {
+		return "", fmt.Errorf("failed to get last user: %w", result.Error)
+	}
+
+	return fmt.Sprintf("%s_%d", name, nextNum), nil
```

Comment:
Funziona, però go sarebbe meglio scriverlo in maniera che il branch più probabile sia sempre quello "pricipale". Considerato che è molto probabile trovare un nome appena abbiano un minimo di utenti scriverei il codice così:
```suggestion
	if result.Error != nil {
		if result.Error != gorm.ErrRecordNotFound {
			return "", fmt.Errorf("failed to get last user: %w", result.Error)
		} else {
			return fmt.Sprintf("%s_1", name), nil
		}
    }

	re := regexp.MustCompile(fmt.Sprintf(`^%s_(\d+)$`, name))
	if matches := re.FindStringSubmatch(lastUser.Alias); len(matches) == 2 {
		if n, err := strconv.Atoi(matches[1]); err == nil {
			nextNum = n + 1
		}
	}
	return fmt.Sprintf("%s_%d", name, nextNum), nil

```
Mi rendo conto di rompere su questioni di stile, però dopo che leggi tanto Go sono tutti scritti così i programmi e ti trovi meglio.